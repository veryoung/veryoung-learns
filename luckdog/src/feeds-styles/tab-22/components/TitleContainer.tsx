/**
 * 包含Title的卡片
 * Created by liquid @2019年10月16日19:26:38
 */
import React from 'react';
import { View } from '@tencent/hippy-react-qb';
import { Title } from '.';
import FeedsProtect from '../../../mixins/FeedsProtect';
import { shouldComponentUpdate, countReRender } from '@tencent/luckbox-react-optimize';

interface Props {
  parent: any;
}

@FeedsProtect.protect
export class TitleContainer extends React.Component<Props> {
  public constructor(props) {
    super(props);
    this.shouldComponentUpdate = shouldComponentUpdate(this, 'TitleContainer');
  }

  public render() {
    countReRender(this, 'TitleContainer');
    const { props } = this;
    const { parent = {} } = props;
    const { itemBean = {} } = parent.props || {};
    const { title } = itemBean;
    return (
      <View style={{ marginTop: 8 }}>
        {title && <Title title={title} parent={parent} />}
        <View
          style={{
            marginHorizontal: 12,
          }}
        >
          {props.children}
        </View>
      </View>
    );
  }
}
