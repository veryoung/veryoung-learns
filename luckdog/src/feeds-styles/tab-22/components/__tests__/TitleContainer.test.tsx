import React from 'react';
import renderer from 'react-test-renderer';

import { TitleContainer } from '../TitleContainer';
import { commonProps } from '@test/mocks/props';

describe('<TitleContainer/>', () => {
  describe('render testing', () => {
    it('render normal', () => {
      const props = {
        ...commonProps,
      };
      const tree = renderer.create(<TitleContainer {...props} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
