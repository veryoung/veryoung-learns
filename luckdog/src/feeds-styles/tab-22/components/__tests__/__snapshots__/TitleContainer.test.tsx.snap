// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleContainer/> render testing render normal 1`] = `
<div
  nativeName="View"
  style={
    Object {
      "marginTop": 8,
    }
  }
>
  <div
    nativeName="View"
    style={
      Object {
        "marginHorizontal": 12,
      }
    }
  />
</div>
`;
